Use mongodb-driver-sync instead of mongo-java-driver (#6375)
quarkusio/quarkus: docs/src/main/asciidoc/mongodb.adoc

    private final Codec<Document> documentCodec;

    public FruitCodec() {

>         this.documentCodec = MongoClientSettings.getDefaultCodecRegistry().get(Document.class);

Reviewer: Do we need a PR for the quickstart too (if there is one)?

Author: Yes, I'll send it today or tomorrow.

As we add some stuff to Hibernate with Panache and MongoDB with Panache lately I will check these two quickstart also to align them with the guides.

This behavior can be disabled by setting the `quarkus.mongodb.health.enabled` property to `false` in your `application.properties`.

== The legacy client

> We didn't include the legacy MongoDB client per default. It contains the old MongoDB Java API (DB, DBCollection,... )

Reviewer: ```suggestion
We don't include the legacy MongoDB client by default. It contains the now retired MongoDB Java API (DB, DBCollection,... )
```

and the `com.mongodb.MongoClient` that is now superseded by `com.mongodb.client.MongoClient`.

> If you want to use the legacy API, you need to include the following dependency inside your `pom.xml`.

Reviewer: ```suggestion
If you want to use the legacy API, you need to add the following dependency to your `pom.xml`:
```
